src/bin/main.ml

```diff
@@ -233,7 +241,45 @@ let start urls sandbox git ~bare =
       )
   ) urls
 
-let start () url sandbox git bare = Lwt_main.run (start url sandbox git ~bare)
+let start () url sandbox git bare auto_push =
+  let start () = start url sandbox git ~bare in
+  Lwt_main.run begin
+    match auto_push with
+    | None        -> start ()
+    | Some remote ->
+      let watch () = match git with
+        | None      ->
+          In_memory_store.repo () >>= fun repo ->
+          In_memory_store.Store.Repo.watch_branches repo (fun _ _ ->
+              Lwt.fail_with "TOTO"
+            )
+        | Some path ->
+          Lazy.force Git_fs_store.listener;
+          let prefix = if sandbox then "." else "" in
+          let path = prefix ^ path in
+          let push () =
+            Logs.debug (fun l -> l "Pushing %s to %s" path remote);
+            let cmd =
+              Lwt_process.shell @@
+              Printf.sprintf "cd %s && git push %s --all" path remote
```

Comment:
The `%s` here looks risky. At least `%S` would work with spaces. Annoying that `Lwt_process` doesn't provide a portable way to set the directory.